src/embind/emval.js

```diff
@@ -195,22 +196,23 @@ var LibraryEmVal = {
 
     for (var i = 0; i < argCount; ++i) {
         functionBody +=
-            "var argType"+i+" = requireRegisteredType(Module['HEAP32'][(argTypes >>> 2) + "+i+"], \"parameter "+i+"\");\n" +
+            "var argType"+i+" = requireRegisteredType({{{ makeGetValue('argTypes', '0', '*') }}}, 'parameter "+i+"');\n" +
             "var arg"+i+" = argType"+i+".readValueFromPointer(args);\n" +
             "args += argType"+i+"['argPackAdvance'];\n";
+            "argTypes += {{{ POINTER_SIZE }}}";
```

Comment:
I don't understand why this does not change behavior for wasm32 - this line is entirely new, and I can't find a `+= 4` or something else that this replaces?

Reply:
If you look at line 199 there are two different "+i" in the old code.

I don't remember why I did this refactoring and I could move this back into the `makeGetValue` expression above.   I think it probably makes more sense there.

Reply:
Actually due to the way this code is constructing code based on strings I don't think we can avoid doing it this way.